knowledge_repo/repositories/folder.py

```diff
@@ -137,18 +144,17 @@ def _kp_get_revisions(self, path):  # slow
         raise NotImplementedError()
 
     def _kp_write_ref(self, path, reference, data, uuid=None, revision=None):
-        path = os.path.join(self.path, path)
-        if os.path.isdir(path):
-            ref_path = os.path.join(path, reference)
+        if os.path.isfile(path):
```

Comment:
@matthewwardrop I put back PKP support PTAL

Reply:
It's not clear to me why checking whether the path is a file is better than checking whether it is a directory, but I have no objection.

You still need the `path = os.path.join(self.path, path)` line, however, otherwise you will be looking for files relative to the current working directory, rather than the repository. `ref_path` should then be returned to just being the join of `path` and `reference`.

Reply:
Here was the issue... 

When running 
```
knowledge_repo --repo test_folder_repo add example.ipynb -p 'projects/example.kp'
```
several references are written to that path.

 For the first reference, path `projects/example.kp` is not a directory (does not exist) and we therefore do right thing and create that directory and write the reference. Then for the next reference the path IS a directory and so we then try to read it as a PKP. Make sense?

Reply:
Yep... That makes a lot of sense :). Once you've sorted out path issue, and then merge it in :).